lib/IRGen/GenClass.cpp

```diff
@@ -212,6 +240,21 @@ namespace {
       return Elements;
     }
 
+    bool isFixedSize() const {
+      return !(ClassHasMissingMembers ||
+               ClassHasResilientMembers ||
+               ClassHasGenericLayout ||
+               ClassHasResilientAncestry ||
+               ClassHasObjCAncestry);
+    }
+
+    bool doesMetadataRequireDynamicInitialization() const {
+      return (ClassHasMissingMembers ||
```

Comment:
I'm pretty sure a Swift-defined class you're emitting can never have missing members, which means this particular bit isn't so interesting.

Reply:
I don't remember if we allow subclassing classes with missing members. If we do, then this can happen during emission too.

Reply:
We do, but is this query ever used in that case? If it's just about the superclass reference, missing members won't matter, because they're only missing from the perspective of this compilation unit.

Reply:
I think so. The ClassLayoutBuilder walks the fields of the superclass too, so ClassHasMissingMembers will be true when emitting the subclass's metadata.

Reply:
Ah, right, because the entire hierarchy might otherwise be non-resilient. Okay.

Reply:
Can you add a test for this case? It can be heavy-handed, i.e. just removing a C struct entirely under certain -Xcc flags.

Reply:
There's already some tests for this, see `class SubButtHolder` in test/IRGen/mixed_mode_class_with_unimportable_fields.swift